launcher/minecraft/mod/Resource.cpp

```diff
@@ -89,6 +106,13 @@ std::pair<int, bool> Resource::compare(const Resource& other, SortType type) con
             if (dateTimeChanged() < other.dateTimeChanged())
                 return { -1, type == SortType::DATE };
             break;
+        case SortType::SIZE: {
+            if (fileinfo().size() > other.fileinfo().size())
+                return { 1, type == SortType::SIZE };
+            if (fileinfo().size() < other.fileinfo().size())
+                return { -1, type == SortType::SIZE };
```

Comment:
This should be changed to always sort dirs first

Reply:
Sorry but I can't find a good way to do this.
Right now this function always returns that a directory is less than a file(regardless of the content of the folder).
So unless I change all the functions to pass the `sortOrder` I don't think that is possible in a clean way(there are a lot of function signatures that need to be changed to achieve this). 
Can we keep it as is?

Reply:
Simply treat directories as larger?

Reply:
Sadly nope. Can you try to implement this, as maybe I do not understand what you really want?
I though you wanted to have the folders on top of the list regardless of how you sort after size. And if I say that the folders are smaller/bigger than files the sorting will respect that and will put them at the end of the list if I sort it desc or asc order.

Reply:
Hmm, I guess treating them smaller is how nautilus does it. That's what I mean then :P

Maybe you could do this by treating folders size (in comparison) as `-[item count]` and files size as... just the number of bytes like it currently is.

[Screencast from 2023-12-20 13-36-02.webm](https://github.com/PrismLauncher/PrismLauncher/assets/57493648/1786fa21-7a60-4046-8fd8-659f17126960)

Reply:
@TheKodeToad feel free to modify this as you see fit.
From my perspective, this PR already does what the issue requested and is complete.
I would make the changes myself but I do not seem to understand exactly what you want.

Reply:
Directories will be sorted by [this](https://stackoverflow.com/a/12240359/7658797) instead which I would consider very unintuitive.

I will try to make this change!

Reply:
actually just to avoid approving code i've directly written here is something which should give you the gist (pseudo-code):
```suggestion
			int sortSize(const QFileInfo& file) {
				if (file.isDir()) {
					QDir dir(file.absoluteFilePath());
					dir.setFilter(QDir::AllEntries | QDir::NoDotAndDotDot);
					return -dir.count();
				}

				return file.size();
			}

			int f1 = sortSize(fileinfo());
			int f2 = sortSize(other.fileinfo());

            if (f1 > f2)
                return { 1, type == SortType::SIZE };
            if (f1 < f2)
                return { -1, type == SortType::SIZE };
```

Reply:
This is how it's sorts using your gist:
![Screenshot_2023-12-22_23-38-02](https://github.com/PrismLauncher/PrismLauncher/assets/42720688/4cfd698d-ecd8-4cf7-bb36-26e4af0167a8)
![Screenshot_2023-12-22_23-38-36](https://github.com/PrismLauncher/PrismLauncher/assets/42720688/898ed115-be69-47d7-a002-73f555dd33a7)

Reply:
Oh xD

That isn't what I intended. oops.

Reply:
That's why I'm saying that if you want to change something here just do a commit yourself

Reply:
Well I would consider this behaviour (and the previous too) broken so marking as unresolved for now

Reply:
you probably want to make the size_info a tuple/pair where the first item is the type (single item / folder) and the second item is the size. tuples of the same type can be lexicographic compared.

Reply:
No need for that: I realized what TheKodeToad wanted here (keep folders at the top regardless of are ascending or descending) so the tuple wouldn't have solved that. Also in the comparing place we have access to the information and we do not need to store it in another variable.